test/e2e/fixture.go

```diff
@@ -124,24 +124,26 @@ func forAllNodes(fn func(nodeName string) error) error {
 	return nil
 }
 
-func exportLogs(tb testing.TB, data *TestData, logsSubDir string, writeNodeLogs bool) {
-	if tb.Skipped() {
-		return
-	}
-	// if test was successful and --logs-export-on-success was not provided, we do not export
-	// any logs.
-	if !tb.Failed() && !testOptions.logsExportOnSuccess {
-		return
+func exportLogs(tb *testing.TB, data *TestData, logsSubDir string, writeNodeLogs bool, testMain bool) {
```

Comment:
I feel a little weird about changing the parameter from `testing.TB` to `*testing.TB` but probably not a big deal for now.
Other changes LGTM.

Reply:
if we don't change testing.TB to a pointer, I don't know what parameter to pass here when making call from testMain.

Reply:
`testing.TB` is the interface common to T and B, so you could try passing an empty T struct.

Reply:
so true! updated. thanks Yongming.